app/components/login.js

```diff
@@ -0,0 +1,33 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { AUTH } from '../constants/urls';
+import { inject as service } from '@ember/service';
+
+export default class LoginComponent extends Component {
+  @service router;
+  @service fastboot;
+  @service login;
+  @tracked authURL = this.generateAuthURL();
+
+  AUTH_URL = this.generateAuthURL();
+
+  generateAuthURL() {
+    let currentURL = this.fastboot.isFastBoot
+      ? this.fastboot.request.protocol +
+        '//' +
+        this.fastboot.request.host +
+        this.fastboot.request.path
+      : window.location.href;
+
```

Comment:
When does this happen?

Reply:
it generates authUrl when user visit login component

Reply:
explain the fastboot part please

Reply:
https://github.com/Real-Dev-Squad/website-www/pull/961#discussion_r1897105622